third_party/blink/renderer/platform/media/web_content_decryption_module_impl.cc

```diff
@@ -154,6 +154,24 @@ void WebContentDecryptionModuleImpl::GetStatusForPolicy(
           kGetStatusForPolicyUMAName));
 }
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+void WebContentDecryptionModuleImpl::GetMetrics(std::string &metrics_results) {
+  auto cdm_context_ref = adapter_->GetCdmContextRef();
+  DCHECK(cdm_context_ref);
+
+  auto* cdm_context = cdm_context_ref->GetCdmContext();
+  DCHECK(cdm_context);
+
+  auto sbdrm = cdm_context->GetSbDrmSystem();
+  DCHECK(SbDrmSystemIsValid(sbdrm));
+
+  int metrics_size = -1;
+  const void* metrics = SbDrmGetMetrics(sbdrm, &metrics_size);
+  DCHECK(metrics_size > 0);
+  metrics_results.assign(static_cast<const char*>(metrics), metrics_size);
```

Comment:
Can we make this function signature `std::string ...::GetMetrics() {` and then here just do `return std::string(static_cast<const char*>(metrics), metrics_size);` ?